WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift

```diff
@@ -1425,6 +1436,14 @@ extension EditableOrderViewModel {
                 return false
             }
         }
+
+        private static func isCouponsError(_ error: Error) -> Bool {
+            if case .unknown(code: "woocommerce_rest_invalid_coupon", _) = error as? DotcomError {
```

Comment:
I see this was used already on a different method, but - related to the comment in the tests - , we seem to accept any sort of Error in this method, but then internally we're requiring a specific DotComError type, which requires us to import `enum Networking.DotcomError` and leak networking details into the business layer. Perhaps we should abstract this?

Reply:
That's an excellent point, thanks for commenting. 

If we want to avoid exposing the `DotcomError` in the WooCommerce target, we would have to parse and convert the `DotcomError` in `Yosemite` to some other error enum public which can be read from WooCommerce. 

The tradeoff of accessing the `DotcomError` in WooCommerce is that we don't have to do that extra task of parsing and duplicating the error, we can just pass along the retrieved one from `Networking`, which is not optimal but more straightforward. This is the general strategy in the app; we read the `DotcomError` in quite a few places through the WooCommerce target.

Reply:
Gotcha! Indeed there seem to be quite a bit of usage of this across the app. I've opened an [issue here](https://github.com/woocommerce/woocommerce-ios/issues/10070) to keep track of it 👍